packages/core/test/Renderer.test.ts

```diff
@@ -77,5 +79,144 @@ describe('utils', () => {
+
+                const ul = new ListNode('UL');
+                const p2 = new VElement('P');
+                p2.append(new CharNode('a'));
+                p2.append(new CharNode('.'));
+                p2.append(new CharNode('a'));
+                ul.append(p2);
+
+                const p3 = new VElement('P');
+                p3.append(new CharNode('a'));
+                p3.append(new CharNode('.'));
+                p3.append(new CharNode('b'));
+                ul.append(p3);
+
+                const h1 = new VElement('H1');
+                h1.append(new CharNode('a'));
+                h1.append(new CharNode('.'));
+                h1.append(new CharNode('c'));
+                ul.append(h1);
+
+                const p4 = new VElement('P');
+                p4.append(new CharNode('a'));
+                p4.append(new CharNode('.'));
+                p4.append(new CharNode('d'));
+                ul.append(p4);
+
+                const ul2 = new ListNode('UL');
+                const p5 = new VElement('P');
+                p5.append(new CharNode('a'));
+                p5.append(new CharNode('.'));
+                p5.append(new CharNode('d'));
+                p5.append(new CharNode('.'));
+                p5.append(new CharNode('a'));
+                ul2.append(p5);
+                ul.append(ul2);
+                motherList.append(ul);
+
+                const p6 = new VElement('P');
+                p6.append(new CharNode('b'));
+                motherList.append(p6);
+
+                const ol = new ListNode('OL');
+                const p7 = new VElement('P');
+                p7.append(new CharNode('b'));
+                p7.append(new CharNode('.'));
+                p7.append(new CharNode('1'));
+                ol.append(p7);
+                const p8 = new VElement('P');
+                p8.append(new CharNode('b'));
+                p8.append(new CharNode('.'));
+                p8.append(new CharNode('2'));
+                ol.append(p8);
+                const h12 = new VElement('H1');
+                h12.append(new CharNode('b'));
+                h12.append(new CharNode('.'));
+                h12.append(new CharNode('3'));
+                ol.append(h12);
+                const p9 = new VElement('P');
+                p9.append(new CharNode('b'));
+                p9.append(new CharNode('.'));
+                p9.append(new CharNode('4'));
+                ol.append(p9);
+                motherList.append(ol);
+
+                const renderer = new Renderer();
+                renderer.render(editor.vDocument, editor.editable);
+                /* eslint-disable prettier/prettier */
+                expect(editor.editable.innerHTML).to.equal([
+                    '<ul>',
+                        '<li>a</li>',
+                        '<li>',
+                            '<ul>',
+                                '<li>a.a</li>',
+                                '<li>a.b</li>',
```

Comment:
My prior question about P remains, since this example introduces no p. What about a list item containing two P elements ?